src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -3542,3 +3545,918 @@ FIXTURE_TEST(test_skipping_compaction_below_start_offset, log_builder_fixture) {
+          },
+          ss::default_priority_class())
+        .get(),
+      std::invalid_argument);
+
+    // Query committed offset of the last batch, expect
+    // no result.
+
+    auto res = log
+                 ->offset_range_size(
+                   nc_summaries.back().last,
+                   storage::log::offset_range_size_requirements_t{
+                     .target_size = 0x10000,
+                     .min_size = 0,
+                   },
+                   ss::default_priority_class())
+                 .get();
+
+    BOOST_REQUIRE_EQUAL(res.last_offset, model::offset{});
+
+    // Query offset out of range to trigger the exception.
+
+    BOOST_REQUIRE_THROW(
+      log
+        ->offset_range_size(
+          model::next_offset(c_summaries.back().last),
+          storage::log::offset_range_size_requirements_t{
+            .target_size = 0x10000,
+            .min_size = 0,
+          },
+          ss::default_priority_class())
+        .get(),
+      std::invalid_argument);
+
+    // Check that the last batch can be measured independently
+    res = log
+            ->offset_range_size(
+              c_summaries.back().base,
+              storage::log::offset_range_size_requirements_t{
+                .target_size = 0x10000,
+                .min_size = 0,
+              },
+              ss::default_priority_class())
+            .get();
+
+    // Only one batch is returned
+    BOOST_REQUIRE_EQUAL(res.last_offset, lstat.committed_offset);
+    BOOST_REQUIRE_EQUAL(
+      res.on_disk_size, c_acc_size.back() - c_prev_size.back());
+
+    // Check that we can measure the size of the log tail. This is needed for
+    // timed uploads.
+    size_t tail_length = 5;
+
+    for (size_t i = 0; i < tail_length; i++) {
+        auto ix_batch = c_summaries.size() - 1 - i;
+        res = log
+                ->offset_range_size(
+                  c_summaries.at(ix_batch).base,
+                  storage::log::offset_range_size_requirements_t{
+                    .target_size = 0x10000,
+                    .min_size = 0,
+                  },
+                  ss::default_priority_class())
+                .get();
+
+        BOOST_REQUIRE_EQUAL(res.last_offset, lstat.committed_offset);
+        BOOST_REQUIRE_EQUAL(
+          res.on_disk_size, c_acc_size.back() - c_prev_size.at(ix_batch));
+    }
+
+    // Check that the min_size is respected
+    res = log
+            ->offset_range_size(
+              c_summaries.back().base,
+              storage::log::offset_range_size_requirements_t{
+                .target_size = 0x10000,
+                .min_size = c_summaries.back().batch_size + 1,
+              },
+              ss::default_priority_class())
+            .get();
+
+    BOOST_REQUIRE_EQUAL(res.last_offset, model::offset{});
+
+#endif
+};
+
+FIXTURE_TEST(test_offset_range_size_incremental, storage_test_fixture) {
+#ifdef NDEBUG
+    // This test attempts to consume the log incrementally using overload
```

Comment:
This is great cover letter for the test. Since these tests are not the most obvious, I and future readers will greatly appreciate a cover letter on all of them.

Reply:
done